compiler/next/lib/uast/Builder.cpp

```diff
@@ -270,6 +271,26 @@ void Builder::Result::updateFilePaths(Context* context, const Result& keep) {
   }
 }
 
+ASTList Builder::flattenTopLevelBlocks(ASTList lst) {
+  ASTList ret;
+
+  std::reverse(lst.begin(), lst.end());
+
+  while (lst.size() > 0) {
+    auto ast = std::move(lst.back());
+    lst.pop_back();
+    if (ast->isBlock()) {
+      for (auto& child : takeChildren(std::move(ast))) {
+        ret.push_back(std::move(child));
+      }
+    } else {
+      ret.push_back(std::move(ast));
+    }
+  }
```

Comment:
This is fine but I'd think it would be simpler if written without `std::reverse`. You can just iterate over the `lst` appending to `ret`. e.g. maybe something along these lines

```
for (auto& elt : lst) {
  if (elt->isBlock()) {
    for (auto& child : takeChildren(std::move(elt))) {
      ret.push_back(std::move(child));
    }
  } else {
    ret.push_back(std::move(elt));
  }
}
lst.clear();
```

Reply:
Ok.